File: src/main/java/upbrella/be/rent/repository/RentRepositoryCustom.java
package upbrella.be.rent.repository;

import org.springframework.data.domain.Pageable;
import upbrella.be.rent.dto.response.HistoryInfoDto;
import upbrella.be.rent.dto.request.HistoryFilterRequest;
import upbrella.be.rent.entity.History;

import java.util.List;

public interface RentRepositoryCustom {

    List<History> findAll(HistoryFilterRequest filter, Pageable pageable);

    List<HistoryInfoDto> findHistoryInfos(HistoryFilterRequest filter, Pageable pageable);

    long countAll(HistoryFilterRequest filter, Pageable pageable);

    List<History> findAllByUserId(long userId);
}

File: src/main/kotlin/upbrella/be/rent/repository/RentRepositoryCustom.kt
package upbrella.be.rent.repository

import org.springframework.data.domain.Pageable
import upbrella.be.rent.dto.request.HistoryFilterRequest
import upbrella.be.rent.dto.response.HistoryInfoDto
import upbrella.be.rent.entity.History

interface RentRepositoryCustom {

    fun findAll(filter: HistoryFilterRequest, pageable: Pageable): List<History>

    fun findHistoryInfos(filter: HistoryFilterRequest, pageable: Pageable): List<HistoryInfoDto>

    fun countAll(filter: HistoryFilterRequest, pageable: Pageable): Long

    fun findAllByUserId(userId: Long): List<History>
}